cfme/cloud/flavor.py

```diff
@@ -210,6 +211,23 @@ def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True, c
         flavor = self.instantiate(name, provider, ram, vcpus, disk, swap, rxtx, is_public)
         return flavor
 
+    def all(self):
+        provider = self.filters.get('provider')  # None if no filter, need for entity instantiation
+        view = navigate_to(self, 'All')
+        result = []
+        for _ in view.entities.paginator.pages():
+            flavors = view.entities.get_all()
+            for flavor in flavors:
+                if provider is not None:
+                    if flavor.data['cloud_provider'] == provider.name:
+                        entity = self.instantiate(flavor.data['name'],
+                                              get_crud_by_name(flavor.data['cloud_provider']))
```

Comment:
Required:
no need to create a provider object using `get_crud_by_name`. use it  directly